pkg/api/controller.go

```diff
@@ -378,6 +384,14 @@ func (c *Controller) LinkPhysicalAddress(w http.ResponseWriter, r *http.Request,
 	writeTime := time.Now()
 	physicalAddress, addressType := normalizePhysicalAddress(repo.StorageNamespace, StringValue(body.Staging.PhysicalAddress))
 
+	// validate token
+	err = c.Catalog.GetAddressToken(ctx, repository, physicalAddress)
+	if err != nil {
+		writeError(w, r, http.StatusInternalServerError, err)
```

Comment:
This should go through handleAPIError. I don't think internal server error is appropriate for trying to link with a token that doesn't exist

Reply:
done